code/modules/awaymissions/corpse.dm

```diff
@@ -36,10 +53,35 @@
 /obj/effect/landmark/corpse/proc/createCorpse() //Creates a mob and checks for gear in each slot before attempting to equip it.
 	var/mob/living/carbon/human/human/M = new /mob/living/carbon/human/human (src.loc)
 	M.real_name = src.name
-	M.death(1) //Kills the new mob
 	M.timeofdeath = timeofdeath
-	if(src.mob_species)
-		M.set_species(src.mob_species)
+	M.gender = pick(MALE, FEMALE)
+	if(!src.mob_species)
+		M.set_species(pickweight(cspecies)) //If no species specified by the mob type, select one at random
+	M.set_species(src.mob_species)
+	M.death(1) //Kills the new mob
+	M.emp_act(1)
+	var/obj/item/organ/external/head/head_organ = M.get_organ("head")
+	if(head_organ)
+		head_organ.r_facial = rand(0,255) //facial coloring
+		head_organ.g_facial = rand(0,255)
+		head_organ.b_facial = rand(0,255)
+		head_organ.r_hair = rand(0,255) //hair coloring
+		head_organ.g_hair = rand(0,255)
+		head_organ.b_hair = rand(0,255)
+	M.s_tone = rand (-185,34) //skin tone - this isn't being updated on the mobs
+	M.r_skin = rand (0,255) //skin coloring
+	M.g_skin = rand (0,255)
+	M.b_skin = rand (0,255)
+	M.r_markings = rand (0,255) //markings coloring, for when this can do other species
+	M.g_markings = rand (0,255)
+	M.b_markings = rand (0,255)
+	head_organ.h_style = random_hair_style(M.gender, head_organ.species.name) // hair style
+	head_organ.f_style = random_facial_hair_style(M.gender, head_organ.species.name) // facial hair style
+	M.update_hair() //update icons
+	M.update_fhair()
+	M.update_markings()
+	M.update_body()

```

Comment:
you can replace `update_hair`, `update_fhair`, `update_markings`, and `update_body` with `regenerate_icons`. This may also get the skin tone to work properly.